src/avt/DBAtts/SIL/CMakeLists.txt

```diff
@@ -4,20 +4,39 @@
 
 #*****************************************************************************
 # Modifications:
+#   Kathleen Biagas, Mon Jun 3, 2024
+#   Modified to use visit_append_list.
 #
 #****************************************************************************
 
-target_sources(avtdbatts PRIVATE
-    avtSIL.C
-    avtSILArray.C
-    avtSILCollection.C
-    avtSILEnumeratedNamespace.C
-    avtSILMatrix.C
-    avtSILNamespace.C 
-    avtSILRangeNamespace.C
-    avtSILRestriction.C
-    avtSILRestrictionTraverser.C
-    avtSILSet.C)
+visit_append_list(
+    NAME   avtdbatts_SOURCES
+    ITEMS  ${CMAKE_CURRENT_SOURCE_DIR}/avtSIL.C
+           ${CMAKE_CURRENT_SOURCE_DIR}/avtSILArray.C
+           ${CMAKE_CURRENT_SOURCE_DIR}/avtSILCollection.C
+           ${CMAKE_CURRENT_SOURCE_DIR}/avtSILEnumeratedNamespace.C
+           ${CMAKE_CURRENT_SOURCE_DIR}/avtSILMatrix.C
+           ${CMAKE_CURRENT_SOURCE_DIR}/avtSILNamespace.C
+           ${CMAKE_CURRENT_SOURCE_DIR}/avtSILRangeNamespace.C
+           ${CMAKE_CURRENT_SOURCE_DIR}/avtSILRestriction.C
+           ${CMAKE_CURRENT_SOURCE_DIR}/avtSILRestrictionTraverser.C
+           ${CMAKE_CURRENT_SOURCE_DIR}/avtSILSet.C)
 
-target_include_directories(avtdbatts PUBLIC .)
+visit_append_list(
+    NAME   avtdbatts_HEADERS
```

Comment:
Remind me... this `_HEADERS` and `_INCLUDES` nomenclature is a BLT thing, right? And, what is the difference...headers are installed includes whereas includes are just internal to the implementation?

Reply:
It is not a BLT thing, it is a work-around for the fact that `blt_add_library` currently requires sending SOURCES list, whereas CMake does not. CMake allows `add_library` call with just a name, and calling `target_sources` later with the list of sources.

Reply:
We decided to make use of the HEADERS (basically considered as 'source files' by CMake) argument to `blt_add_library` so that headers are tracked just like source files for changes, etc, and possibly used for some of the blt utilities we will make use of in the future.
INCLUDES inform CMake about  building not only the library being created but other targets that use the library.